Sources/Sentry/SentryNSURLSessionTaskSearch.m

```diff
@@ -38,10 +38,9 @@ @implementation SentryNSURLSessionTaskSearch
     NSURLSessionConfiguration *configuration =
         [NSURLSessionConfiguration ephemeralSessionConfiguration];
     NSURLSession *session = [NSURLSession sessionWithConfiguration:configuration];
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wnonnull"
-    NSURLSessionDataTask *localDataTask = [session dataTaskWithURL:nil];
-#pragma clang diagnostic pop
+
+    NSURLSessionDataTask *localDataTask = [session dataTaskWithURL:[NSURL new]];
```

Comment:
What's the wide effect of using an instance of nsurl? Since it points nowhere, I wonder.
We're missing a test or e2e that reproduces the crash so we can validate the fox works, and avoid regressions

Reply:
I fully agree with @bruno-garcia. According to our customer

> we have another library that is also trying to track the network session... once i disable that, i no longer see crash on Sentry end

I'm waiting for their response on which library it is they are using.

Reply:
> What's the wide effect of using an instance of nsurl?

None, we don't even use the `localDataTask` as a session, that's why using `nil` also works, we are just interested in retrieving `localDataTask` real class.

But if the user is using another library to track the network session, that other class is probably waiting for a URL parameter as it shouldn't be `nil` according to the documentation.

Reply:
According to our customer, they get this error when using https://github.com/DataDog/dd-sdk-ios v1.6.0 in combination with our SDK.

Reply:
Using DataDog with their networking tracking causes the crash. 
The change proposed on this PR fixed the error.

DataDog is written in Swift, and if we take a look on their [NSSession swizzling implementation](https://github.com/DataDog/dd-sdk-ios/blob/e45f38046b275a7f7f6d26505d8a98d003f98d31/Sources/Datadog/URLSessionAutoInstrumentation/URLSessionSwizzler.swift#L241-253) line 42 
```swift
typealias Signature = @convention(block) (URLSession, URL) -> URLSessionDataTask
```
`URL` is non nullable. When the runtime bridges  or nil `NSURL` to `URL` it crashes.

Another solution is to initialize SentrySDK first.

Reply:
`m`: Can we please add a comment on why we use `[NSUrl new]` and why we must not use `nil`?

Reply:
`m`: Can we maybe add a test to reproduce this bug @brustolin?

Reply:
> `m`: Can we maybe add a test to reproduce this bug @brustolin?

I don't see a proper way of testing this, unless we swizzle NSURLSession in test just to see if `dataTaskWithURL:` was called with a proper argument. I don't like this solution.